guides/docs/testing/testing_controllers.md

```diff
@@ -431,7 +435,7 @@ def show(conn, %{"id" => id}) do
     nil ->
       conn
       |> put_status(:not_found)
-      |> json(%{error: "User not found"})
+      |> json(%{errors: "User not found."})
```

Comment:
It doesn't make sense to say errors and then return a string. Maybe better to revert the change to this line for now?

Reply:
@josevalim it's only matching what is in the test:

```elixir
test "Responds with a message indicating user not found", %{conn:  conn} do
  response =
    conn
    |> get(user_path(conn, :show, -1 ))
    |> json_response(404)

  expected = %{ "errors" => "User not found." }

  assert response == expected
end
```

I can update both to return only a string message, what you think?

Reply:
I see. Can we fix the test in this case? We should also remove the spaces after %{ and before }. Tks!

Reply:
I just realize that, by [JSON definition](http://json.org/), the API must either return a Javascript _Object_ or an _Array_. If we move to returning just *text* we may need to also change the text of the guide.

I'd rather then make that change in a separate PR, so I can review the text together with the code and make sure it makes sense.

This one still stay "weird" but makes the tests green 😄